packages/react/slot/src/Slot.tsx

```diff
@@ -5,7 +5,7 @@ import { composeRefs } from '@radix-ui/react-compose-refs';
  * Slot
  * -----------------------------------------------------------------------------------------------*/
 
-interface SlotProps {
+interface SlotProps extends React.HTMLAttributes<HTMLElement> {
```

Comment:
I'm confused about this, wasn't that the whole point that DOM props should go on the DOM element (child of `Slot`) rather than `Slot`?

Reply:
I see, that's interesting. Do we foresee a lot of these use cases? Generally `Slot` wouldn't be used publicly but I can see in this case what he was trying to do. I'm just curious why we didn't want to do it before with `as` cause technically it was the same thing, you might have want your internal `Button` to have some attributes on it.

Reply:
Either way, we need this same change for our internal use. The only reason we didn't notice this was because I did `const Comp: any = asChild ? Slot : node` in `Primitive` 🙈 . If I made that all a bit stricter it probably would have caught this same issue for us.

We didn't do it before with `as` because we were trying to encourage them to put the props on the thing that actually renders the DOM element (which would be the child). They could have put them on the `Slot` and it would have worked fine... it was just more of a DX decision I believe.